platform/android/api/api.cpp

```diff
@@ -34,11 +34,21 @@
 #include "jni_singleton.h"
 
 #include "core/config/engine.h"
+#include "platform/android/rendering_native_surface_android.h"
 
 #if !defined(ANDROID_ENABLED)
 static JavaClassWrapper *java_class_wrapper = nullptr;
 #endif
 
+void register_core_android_api() {
+#if defined(ANDROID_ENABLED)
+	GDREGISTER_ABSTRACT_CLASS(RenderingNativeSurfaceAndroid);
```

Comment:
Will a stub class for `RenderingNativeSurfaceAndroid` be registered when _not_ on Android? Ideally, we'd want a class the same name and all the same methods available on all platforms, but only a version that actually works when on Android.

We do a similar thing for `JNISingleton`, `JavaClassWrapper`, and `JavaClass` - those are always registered (although, sometimes in a different way) on all platforms, but when not on Android, their real functionality is removed.

I haven't looked at the other platforms, but we'd want to do the same for them as well. Or, alternatively, we could just expose the base class and a factory method, and then developers never need to interact with the classes for the individual platforms, just the base class (which is sort of like what we do for `DisplayServer`).

The point is just that we don't want classes to appear/disappear when on different platforms. Otherwise, developers get errors depending on which platform they opened their project up on, just because they refer to the class name in their GDScript.